File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function(cost) {
  const costToStep = _.memoize(
    /**
     * @param {number} index
     * @returns {number}
     */
    function findCost(index) {
      if (index < 2) {
        return cost[index];
      }

      return cost[index] + Math.min(costToStep(index - 2), costToStep(index - 1));
    }
  );
  
  return Math.min(costToStep(cost.length - 2), costToStep(cost.length - 1));
};

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.go
func minCostClimbingStairs(cost []int) int {
    n := len(cost)
    
    for i := 2; i < n; i++ {
        cost[i] += min(cost[i - 1], cost[i - 2])
    }
    
    return min(cost[n - 1], cost[n - 2])
}

// min returns the lesser of two integers
func min(a, b int) int {
    if a < b {
        return a
    }
    return b
}